nautobot_ssot/metrics.py

```diff
@@ -0,0 +1,156 @@
+"""Nautobot SSoT framework level metrics."""
+from django.conf import settings
+from prometheus_client.core import GaugeMetricFamily
+from nautobot.extras.models.jobs import Job
+from nautobot_ssot.models import Sync
+
+PLUGIN_SETTINGS = settings.PLUGINS_CONFIG.get("nautobot_ssot", {})
+
+
+def metric_ssot_jobs():
+    """Extracts duration of latest SSoT Job run.
+
+    Yields:
+        GaugeMetricFamily: Prometheus Metrics
+    """
+    ssot_job_durations = GaugeMetricFamily(
+        "nautobot_ssot_duration_microseconds",
+        "Nautobot SSoT Job Phase Duration in microseconds",
+        labels=["phase", "job"],
+    )
+
+    for job in Job.objects.filter(slug__icontains="ssot"):
```

Comment:
As long as the name of the project that the Job comes from follows our standard of having SSoT in the name then it should be fine. From what I saw the slug is essentially the full python path to the Job so should contain `ssot` as long as they are following the standard. Unfortunately, I couldn't think of another method of determining SSoT-specific Jobs. If you can think of a method I'm open to changing this.

Reply:
This should work:

```
[sync["job_result__job_model__name"] for sync in Sync.objects.values("job_result__job_model__name").order_by().distinct()]
```

Reply:
This only works if there has already been a Sync attempted for the Job so a Sync object is created. My method allows for all SSoT Jobs regardless of whether they've been ran yet to show metrics.

Reply:
@Kircheneer: Do you still want to use Sync objects or is looking for `ssot` in the Job slug acceptable? I want to get this PR merged in if possible.

Reply:
Does it make sense to report phase duration for jobs that haven't yet run (i.e. that don't have `Sync` object instances)? I don't think so, and therefore I think it should be alright if we went by the `Sync` object route

Reply:
Typically, I'd agree with you, but I could see a new user being confused as to why they don't see the metric when they first enable them and it might be confusing that they are required to complete a Sync before the metrics work.